manticore/ethereum/__init__.py

```diff
@@ -672,6 +672,24 @@ def solidity_create_contract(self, source_code, owner, name=None, contract_name=
             return None
         return contract_account
 
+    def get_nonce(self, address):
+        # type forgiveness:
+        address = int(address)
+        # get all states containing this address:
+        states = tuple(state for state in self.running_states if address in state.platform)
+        if not states:
+            raise NoAliveStates("There are no alive states containing address %x" % address)
+        if len(states) == 1:
+            return states[0].platform.get_nonce(address)
+        # if there are multiple states with this address, ensure that they all currently have the same nonce:
+        existing_nonces = set(state.platform.get_nonce(address) for state in states)
+        if len(existing_nonces) != 1:
+            raise EthereumError("Cannot increase the nonce of address %x because it exists in multiple states with different nonces" % address)
+        ret = None
+        for state in states:
+            ret = state.platform.get_nonce(address)
```

Comment:
Is that fine that we iterate over many states but only take a result of last state?

Maybe we should just do `return states[-1].platform.get_nonce(address)`?
Or maybe we should have it as:
```python
for state in states:
    nonce = state.platform.get_nonce(address)
    if nonce is not None:
        return ret
```